solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -1828,13 +1823,17 @@ private int fetchPackets(FastInputStream fis) throws Exception {
             throw new ReplicationHandlerException("User aborted replication");
           }
           long checkSumServer = -1;
+
           fis.readFully(intbytes);
+
           // read the size of the packet
           int packetSize = readInt(intbytes);
           if (packetSize <= 0) {
-            log.warn("No content received for file: {}", fileName);
+            fis.read(); // read till end-of-file
```

Comment:
This line does not do what the comment says.  Regardless, shouldn't we *already* be at EOF?

Reply:
See `org.apache.solr.common.util.Utils#readFully` instead